src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -5,61 +5,81 @@ import akka.http.scaladsl.Http
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
 import akka.http.scaladsl.model.HttpMethods._
 import akka.http.scaladsl.model.headers._
-import akka.http.scaladsl.model.StatusCodes._
 import akka.http.scaladsl.model._
-import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
 import akka.stream.ActorMaterializer
+import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.model.ErrorReportJsonSupport._
 import org.broadinstitute.dsde.workbench.model.WorkbenchIdentityJsonSupport._
-import org.broadinstitute.dsde.workbench.model.{WorkbenchEmail, UserInfo}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, UserInfo, WorkbenchEmail}
 import org.broadinstitute.dsde.workbench.util.health.StatusJsonSupport._
-import org.broadinstitute.dsde.workbench.util.health.SubsystemStatus
+import org.broadinstitute.dsde.workbench.util.health.StatusCheckResponse
 
 import scala.concurrent.{ExecutionContext, Future}
 
-case class CallToSamFailedException(uri: Uri, status: StatusCode, msg: String)
-  extends LeoException(s"Call to Sam endpoint [${uri.path}] failed with status $status. Message: $msg", status)
+case class CallToSamFailedException(uri: Uri, status: StatusCode, msg: Option[String])
+  extends LeoException(s"Call to Sam endpoint [${uri.path}] failed with status $status.${msg.map(m => s" Message: $m").getOrElse("")}", status)
 
 /**
   * Created by rtitle on 10/16/17.
   */
-class HttpSamDAO(val baseSamServiceURL: String)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends SamDAO {
+class HttpSamDAO(val baseSamServiceURL: String)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends SamDAO with LazyLogging {
   private val samServiceURL = baseSamServiceURL
 
   val http = Http(system)
 
-  override def getStatus(): Future[SubsystemStatus] = {
+  override def getStatus(): Future[StatusCheckResponse] = {
```

Comment:
While in this class, I noticed a bug in this method. Sam /status is returning a `StatusCheckResponse`, but we were unmarshalling the entity as a `SubsystemStatus`. This actually _worked_ due to an accident in spray-json deserialization (both structures contain an `ok` boolean field and all other fields are optional), but the types were actually incorrect. This changes it to deserialize a `StatusCheckResponse` from Sam.

Reply:
Def a 🤦‍♂️ moment